docker/docker_image_dest.go

```diff
@@ -345,8 +345,12 @@ func (d *dockerImageDestination) TryReusingBlobWithOptions(ctx context.Context,
 
 		// Sanity checks:
 		if reference.Domain(candidateRepo) != reference.Domain(d.ref.ref) {
-			logrus.Debugf("... Internal error: domain %s does not match destination %s", reference.Domain(candidateRepo), reference.Domain(d.ref.ref))
-			continue
+			logrus.Debugf("Found blob in a different registry: domain %s with different destination %s", reference.Domain(candidateRepo), reference.Domain(d.ref.ref))
+			// Found following blob in different registry but
+			// we need to check blob presence against the registry
+			// where we are planning to push, hence switch back the
+			// candidate repo to the one where we are planning to push.
+			candidateRepo, _ = parseBICLocationReference(types.BICLocationReference{Opaque: string(d.ref.ref.Name())})
```

Comment:
https://github.com/containers/image/pull/1461/files#r802860796

Reply:
I think you meant `candidateRepo = d.ref.ref`

Reply:
My mistake.  Actually it should be `reference.TrimNamed(d.ref.ref)` to be a _repo_ (though it might happen work just the same in practice right now from a quick look, but let’s not be imprecise about variable semantics).